### PR TITLE
Create github workflow for building and deploying the site

### PR DESCRIPTION
**Pull request by adamjonas**

### CODE REVIEW

File: .github/workflows/adocs-build.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    runs-on: ubuntu-22.04
+    name: make; mv ./build/btcphilosophy.html ./build/index.html
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+    - name: Get build container
+      id: adocbuild
+      uses: tonynv/asciidoctor-action@master
+      with:
+          program: "make; mv ./build/btcphilosophy.html ./build/index.html"
```
Comment (kallerosenbaum) [inline .github/workflows/adocs-build.yml:18]: Is it possible to install imagemagick (`apt install imagemagick`) prior to running `make`? That would make the images resized to more appropriate sizes. 

See https://github.com/bitcoin-dev-philosophy/btcphilosophy/blob/f87b81e93f162aac6896bdbbc5dd69dc20f747df/Makefile#L52-L57

**Comment by willcl-ark** [comment]
@adamjonas I've added a commit which builds with ImageMagick. It needed me to modify the Dockerfile inside of the asciidoc step so I had to clone the whole action, but it will still stay up-to-date etc. so should be fine.

The commit is here: https://github.com/bitcoin-dev-philosophy/btcphilosophy/commit/b2de444805d78c343d728e331d75b3ce2704372f

You should be able to cherry-pick it directly onto your branch with:

```bash
git switch add-adocs-build

git add remote will https://github.com/willcl-ark/btcphilosophy.git
git fetch will
git cherry-pick b2de444

# optional
git remote remove will
```

**Comment by kallerosenbaum** [comment]
Not sure I'm following this, but I'll merge and see what happens...

**Comment by kallerosenbaum** [comment]
Ok, didn't quite work. The actual build went fine it seems, but some issue with an auth token: https://github.com/bitcoin-dev-philosophy/btcphilosophy/actions/runs/4347897369/jobs/7595766103

**Comment by willcl-ark** [comment]
Strange... this seems to imply that it should work automagically:

> For newbies of GitHub Actions: Note that the GITHUB_TOKEN is NOT a personal access token. A GitHub Actions runner automatically creates a GITHUB_TOKEN secret to authenticate in your workflow. So, you can start to deploy immediately without any configuration.

**Comment by willcl-ark** [comment]
Perhaps the solution is [here](https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-first-deployment-with-github_token)?

**Comment by kallerosenbaum** [comment]
Thanks, but I can't find the setting referred to in the link you posted. Maybe I have the wrong role for this and only @adamjonas can do this?

**Comment by kallerosenbaum** [comment]
Yeah, I don't have access to the settings of this repo. So I guess @adamjonas has to fiddle with this :-P

**Comment by adamjonas** [comment]
Believe #82 got this working. This would have gone much longer without your @willcl-ark, thanks.

@kallerosenbaum I believe I got the permissions squared away for you to see everything I do.

**Comment by kallerosenbaum** [comment]
Thanks @adamjonas
Thanks @willcl-ark

**Pull request merged**